cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala

```diff
@@ -62,18 +63,34 @@ class OpenAIPrompt(override val uid: String) extends Transformer
     set(postProcessingOptions, v.asScala.toMap)
 
   val dropPrompt = new BooleanParam(
-    this, "dropPrompt", "whether to drop the column of prompts after templating")
+    this, "dropPrompt", "whether to drop the column of prompts after templating (when using legacy models)")
 
   def getDropPrompt: Boolean = $(dropPrompt)
 
   def setDropPrompt(value: Boolean): this.type = set(dropPrompt, value)
 
+  val dropMessages = new BooleanParam(
+    this, "dropMessages", "whether to drop the column of messages after templating (when using gpt-4 or higher)")
+
+  def getDropMessages: Boolean = $(dropMessages)
+
+  def setDropMessages(value: Boolean): this.type = set(dropMessages, value)
+
+  val systemPrompt = new Param[String](
+    this, "systemPrompt", "The initial system prompt to be used.")
+
+  def getSystemPrompt: String = $(systemPrompt)
+
+  def setSystemPrompt(value: String): this.type = set(systemPrompt, value)
+
   setDefault(
     postProcessing -> "",
     postProcessingOptions -> Map.empty,
     outputCol -> (this.uid + "_output"),
     errorCol -> (this.uid + "_error"),
     dropPrompt -> true,
+    dropMessages -> true,
+    systemPrompt -> "You are an AI Chatbot. Only respond with a completion.",
```

Comment:
```suggestion
    systemPrompt -> "You are an AI chatbot who wants to answer user's questions. Follow their instructions carefully and be breif if they dont say otherwise.",
```